sa4u_z3/main.py

```diff
@@ -261,71 +270,76 @@ def main():
 
     start = time.time()
     count = 0
-    for tu in translation_units(compilation_database, analysis_dir):
-        if analysis_dir:
-            serialized_tu = read_tu(analysis_dir, tu)
-            modified_time = os.path.getmtime(tu.spelling)
-            if serialized_tu.serialization_time >= modified_time:
-                log(
-                    LogLevel.INFO,
-                    f'restoring {tu.spelling} from previous state...'
-                )
+
+    while(True):
+        for tu in translation_units(compilation_database, analysis_dir):
+            if analysis_dir:
```

Comment:
I was trying to move cache logic out of this method, but forgot to finish the thought while I was racing for the ASE deadline. Can you add a comment like "TODO: finish moving cache logic into translation_units` here?